.circleci/config.yml

```diff
@@ -0,0 +1,71 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: circleci/python:2.7
+
+    steps:
+      - checkout
+      - run:
+          name: Install package
+          command: |
+            pip install virtualenv
+            virtualenv venv
+            . venv/bin/activate
+            python setup.py install
+
+      - run:
+          name: Run tests
+          command: |
+            . venv/bin/activate
+            python setup.py nosetests
+
+  deploy:
+    docker:
+      - image: circleci/python:2.7
+
+    steps:
+      - checkout
+      - run:
+          name: Create distribution
+          command: |
+            BRANCH=${CIRCLE_BRANCH#*/}
+
+            # if the release has a tag, create a source distribution
+            if [[ ! -z $CIRCLE_TAG ]]; then
+              python setup.py sdist
+
+            # if the release has no tag, is on the develop branch, and is not part of a PR, tag it dev
+            elif [[ $BRANCH == develop ]] && [[ -z $CIRCLE_PULL_REQUEST ]]; then
+              python setup.py egg_info --tag-build=dev${CIRCLE_BUILD_NUM} sdist
+
+            # otherwise tag it ci
+            else
+              python setup.py egg_info --tag-build=ci${CIRCLE_BUILD_NUM} sdist
+            fi
+
+      - run:
+          name: Huck distribution to repo
+          command: |
+            ssh-keyscan -H $REPO_HOST >> ~/.ssh/known_hosts
+            chmod 600 ~/.ssh/known_hosts
+
+            scp dist/*.tar.gz $SSH_USER@$REPO_HOST:/tmp/
+            ssh -t $SSH_USER@$REPO_HOST "cd /data/repo/pip/prod/ && ls /tmp/ | grep phantom | xargs -I '{}' sudo mv /tmp/'{}' . && sudo su ubuntu -c 'dir2pi -n .'"
+
+workflows:
+  version: 2
+  phantom_snap_workflow:
+    jobs:
+      - build:
+          filters:
+            tags:
```

Comment:
Should this be `branches` instead of `tags`? I'm not seeing the unit tests run on any of the builds in CircleCI.

Reply:
As far as I understand it, the tags filter is only there because it's a transitive dependency of deploy - if a job requires tag checks, all the previous jobs also require tag checks. The job should still trigger on any push. I modeled (read: stole) it off of the Gandalf Circle Ci config, but it might be possible to eliminate tag and branch workflow matching entirely since there's logic in the deploy job to work with different tags. Let me try that right quick.

Reply:
I see the tests being run here: https://circleci.com/gh/istresearch/phantom-snap/16

Each step in a workflow is considered its own build, so you have to click the workflow link near the top to see all the different workflow steps